src/SpectatorView.Unity/Assets/SpectatorView/Scripts/MarkerDetection/QRCode/QRCodesManager.cs

```diff
@@ -1,302 +1,316 @@
+                            newMatrix.M13 = -newMatrix.M13;
+                            newMatrix.M23 = -newMatrix.M23;
+                            newMatrix.M43 = -newMatrix.M43;
+
+                            newMatrix.M31 = -newMatrix.M31;
+                            newMatrix.M32 = -newMatrix.M32;
+                            newMatrix.M34 = -newMatrix.M34;
+
+                            System.Numerics.Vector3 winrtScale;
+                            System.Numerics.Quaternion winrtRotation;
+                            System.Numerics.Vector3 winrtTranslation;
+                            System.Numerics.Matrix4x4.Decompose(newMatrix, out winrtScale, out winrtRotation, out winrtTranslation);
+
+                            var translation = new Vector3(winrtTranslation.X, winrtTranslation.Y, winrtTranslation.Z);
+                            var rotation = new Quaternion(winrtRotation.X, winrtRotation.Y, winrtRotation.Z, winrtRotation.W);
+                            location = Matrix4x4.TRS(translation, rotation, Vector3.one);
+
+                            return true;
+                        }
+                        else
+                        {
+                            Debug.LogWarning("QRCode location unknown or not yet available.");
+                            return false;
+                        }
+                    }
+                    else
+                    {
+                        Debug.LogWarning("Failed to obtain coordinate system for application");
+                        return false;
+                    }
+                }
+                catch(Exception e)
+                {
+                    Debug.LogWarning($"Note: TryGetLocationForQRCode needs to be called from main thread: {e}");
+                    return false;
+                }
+            }
+            else
+            {
+                Debug.LogWarning("Failed to obtain coordinate system for QRCode");
+                return false;
+            }
+        }
+#endif // WINDOWS_UWP
+
+        public Guid GetIdForQRCode(string qrCodeData)
+        {
+            lock (qrCodesList)
+            {
+                foreach (var ite in qrCodesList)
+                {
+                    if (ite.Value.Code == qrCodeData)
+                    {
+                        return ite.Key;
+                    }
+                }
+            }
+            return new Guid();
+        }
+
+        public IList<QRCodesTrackerPlugin.QRCode> GetList()
+        {
+            lock (qrCodesList)
+            {
+                return new List<QRCodesTrackerPlugin.QRCode>(qrCodesList.Values);
+            }
+        }
+
+        void Awake()
+        {
+            IsTrackerRunning = false;
+        }
+
+        protected async void Start()
+        {
+            if (AutoStartQRTracking)
+            {
+                await StartQRTracking();
+            }
+        }
+
+        public async Task<QRTrackerStartResult> StartQRTracking()

```

Comment:
>StartQRTracking [](start = 48, length = 15)

It looks like if you call StartQRTracking (which is async) and then call StopQRTracking before the "await cap.RequestAccessAsync" finishes, the end result would be that the tracker would be running.

I think to make this safe to that kind of timing, you need to separate the state of "IsTrackerRunning" from "ShouldTrackerBeRunning". Then StopQRTracking can always set a state of ShouldTrackerBeRunning to false, and StartQRTracking can check for ShouldTrackerBeRunning after any async call it makes to ensure that it should continue to run the rest of the start sequence